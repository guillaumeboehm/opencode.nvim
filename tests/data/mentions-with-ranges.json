[
  {
    "properties": {},
    "type": "server.connected"
  },
  {
    "properties": {
      "info": {
        "role": "user",
        "time": {
          "created": 1760312301247
        },
        "id": "msg_9daca16bf0017x95VD45mw3k8Q",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.updated"
  },
  {
    "properties": {
      "part": {
        "type": "text",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "text": "when @lua/opencode/ui/streaming_renderer.lua renders a diff, only the first character, the + has the extmark. the rest of the line doesn't seem to have the highlight?\n\nhere's an example from @tests/data/planning.json\n\n**ï‘ˆ  edit** `diff-test.txt`\n\n```txt\n-this is a string\n+this is a great string\n\n```\n\nthe - and the + are highlighted in the right color but none of the text is. any ideas?",
        "id": "prt_9daca16bf0022KbzVMI7i65MRT",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "synthetic": true,
        "type": "text",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "text": "Called the Read tool with the following input: {\"filePath\":\"/Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/streaming_renderer.lua\"}",
        "id": "prt_9daca16c2001opnx6HTUdEAJaV",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "synthetic": true,
        "type": "text",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "text": "test",
        "id": "prt_9daca16c2002SJNzz49CNIFyyl",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "source": {
          "path": "/Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/streaming_renderer.lua",
          "type": "file",
          "text": {
            "value": "@lua/opencode/ui/streaming_renderer.lua",
            "end": 43,
            "start": 5
          }
        },
        "type": "file",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "url": "file:///Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/streaming_renderer.lua",
        "filename": "lua/opencode/ui/streaming_renderer.lua",
        "id": "prt_9daca16c2003hMotbE5rccrYoN",
        "mime": "text/plain",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "synthetic": true,
        "type": "text",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "text": "Called the Read tool with the following input: {\"filePath\":\"/Users/cam/Dev/neovim-dev/opencode.nvim/tests/data/planning.json\"}",
        "id": "prt_9daca16c3001bZ9uR15m68W1TM",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "synthetic": true,
        "type": "text",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "text": "test",
        "id": "prt_9daca16c30024lnVaBquSAsD4l",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "source": {
          "path": "/Users/cam/Dev/neovim-dev/opencode.nvim/tests/data/planning.json",
          "type": "file",
          "text": {
            "value": "@tests/data/planning.json",
            "end": 215,
            "start": 191
          }
        },
        "type": "file",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "url": "file:///Users/cam/Dev/neovim-dev/opencode.nvim/tests/data/planning.json",
        "filename": "tests/data/planning.json",
        "id": "prt_9daca16c3003EkkIdX54sm9uix",
        "mime": "text/plain",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "synthetic": true,
        "type": "text",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "text": "Called the Read tool with the following input: {\"filePath\":\"/Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/session_formatter.lua\"}",
        "id": "prt_9daca16c2004dgt3o9FT1BUWPZ",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "synthetic": true,
        "type": "text",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "text": "test",
        "id": "prt_9daca16c2005WSv0B6gfUZmoVK",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "type": "file",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "url": "file:///Users/cam/Dev/neovim-dev/opencode.nvim/lua/opencode/ui/session_formatter.lua",
        "filename": "lua/opencode/ui/session_formatter.lua",
        "id": "prt_9daca16c2006W0Z6M3NVKB8rQv",
        "mime": "text/plain",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  },
  {
    "properties": {
      "part": {
        "synthetic": true,
        "type": "text",
        "messageID": "msg_9daca16bf0017x95VD45mw3k8Q",
        "text": "{\"context_type\":\"diagnostics\",\"content\":\"Found 2 errors:\\n Line 607: Undefined field `callID`.\\n Line 179: Undefined field `info`.\"}",
        "id": "prt_9daca16bf0037QOQWJXBLwJimF",
        "sessionID": "ses_62582b05affe1Z3mvuBraNHkzT"
      }
    },
    "type": "message.part.updated"
  }
]
