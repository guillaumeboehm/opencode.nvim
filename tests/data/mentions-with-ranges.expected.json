{"extmarks":[[1,2,0,{"virt_text":[["▌󰭻 ","OpencodeMessageRoleUser"],[" "],["USER","OpencodeMessageRoleUser"],["","OpencodeHint"],[" (2025-10-12 23:38:21)","OpencodeHint"],[" [msg_9daca16bf0017x95VD45mw3k8Q]","OpencodeHint"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":false,"virt_text_win_col":-3,"priority":10,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[2,3,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[3,4,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[4,4,5,{"ns_id":3,"end_row":4,"hl_group":"OpencodeMention","end_col":44,"priority":1000,"hl_eol":false,"right_gravity":true,"end_right_gravity":false}],[5,5,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[6,6,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[7,6,23,{"ns_id":3,"end_row":6,"hl_group":"OpencodeMention","end_col":48,"priority":1000,"hl_eol":false,"right_gravity":true,"end_right_gravity":false}],[8,7,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[9,8,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[10,9,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[11,10,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[12,11,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[13,12,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[14,13,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[15,14,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[16,15,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[17,16,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[18,17,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[19,18,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[20,19,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[21,20,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[22,21,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}],[23,22,0,{"virt_text":[["▌","OpencodeMessageRoleUser"]],"virt_text_pos":"win_col","virt_text_repeat_linebreak":true,"virt_text_win_col":-3,"priority":4096,"virt_text_hide":false,"right_gravity":true,"ns_id":3}]],"lines":["","----","","","when @lua/opencode/ui/streaming_renderer.lua renders a diff, only the first character, the + has the extmark. the rest of the line doesn't seem to have the highlight?","","here's an example from @tests/data/planning.json","","**  edit** `diff-test.txt`","","```txt","-this is a string","+this is a great string","","```","","the - and the + are highlighted in the right color but none of the text is. any ideas?","","[lua/opencode/ui/streaming_renderer.lua](lua/opencode/ui/streaming_renderer.lua)","","[tests/data/planning.json](tests/data/planning.json)","","[lua/opencode/ui/session_formatter.lua](lua/opencode/ui/session_formatter.lua)",""],"timestamp":1761108491,"actions":[]}